File: docs/examples/go/add_whitelist.go
// Command add_whitelist creates or updates a whitelist record.
package main

import (
    "bytes"
    "encoding/json"
    "fmt"
    "io"
    "net/http"
)

const whitelistAdminToken = "<token>"

func main() {
    payload := map[string]string{
        "ip":          "192.168.0.10",
        "description": "VPN gateway",
    }
    body, _ := json.Marshal(payload)

    req, _ := http.NewRequest(http.MethodPost, baseURL+"/settings/whitelist", bytes.NewReader(body))
    req.Header.Set("Content-Type", "application/json")
    req.Header.Set("Authorization", "Bearer "+whitelistAdminToken)

    resp, err := http.DefaultClient.Do(req)
    if err != nil {
        panic(err)
    }
    defer resp.Body.Close()

    if resp.StatusCode != http.StatusCreated {
        data, _ := io.ReadAll(resp.Body)
        panic(fmt.Sprintf("unexpected status %d: %s", resp.StatusCode, data))
    }

    var record map[string]any
    if err := json.NewDecoder(resp.Body).Decode(&record); err != nil {
        panic(err)
    }

    fmt.Println("Saved whitelist entry:", record)
}

File: docs/examples/python/add_whitelist.py
"""Add or update an IP address in the whitelist."""

import requests

BASE_URL = "http://localhost:3000/api"
TOKEN = "<token>"


def add_whitelist() -> None:
    payload = {"ip": "192.168.0.10", "description": "VPN gateway"}
    response = requests.post(
        f"{BASE_URL}/settings/whitelist",
        json=payload,
        headers={"Authorization": f"Bearer {TOKEN}"},
        timeout=10,
    )
    response.raise_for_status()
    print("Saved whitelist record:", response.json())


if __name__ == "__main__":
    add_whitelist()
